packages/utils/src/index.ts

```diff
@@ -69,34 +71,33 @@ export const unique = <T>(arr: T[]): T[] => [...new Set(arr)];
 export const intersection = <T>(array1: T[], array2: T[]) =>
   unique(array1.filter(value => array2.includes(value)));
 
-export const pick = <T>(obj: Record<string, T>, keys: string[]): Record<string, T> =>
-  keys.reduce((acc, key) => (key in obj ? { ...acc, [key]: obj[key] } : acc), {});
+export const pick = <T, K extends keyof T>(obj: T, keys: K[]) =>
+  keys.reduce(
+    (acc, key) => (key in obj ? { ...acc, [key]: obj[key] } : acc),
+    {} as { [P in K]: T[P] }
+  );
 
-export const omitBy = <T>(obj: Record<string, T>, func: (key: string) => boolean) =>
+export const omitBy = <T>(obj: T, func: (key: keyof T) => boolean) =>
   pick(
     obj,
-    Object.keys(obj).filter(key => !func(key))
-  );
+    (Object.keys(obj) as (keyof T)[]).filter(key => !func(key))
+  ) as Partial<T>;
 
-export const omit = <T>(obj: Record<string, T>, keys: string[]) =>
-  omitBy(obj, value => keys.includes(value));
+export const omit = <T, K extends keyof T>(obj: T, keys: K[]) =>
+  omitBy(obj, value => keys.includes(value as K)) as Omit<T, K>;
 
 // [{ k1: v1, k2: v2, ...}, { k3: v3, k4: v4, ...}, ...] => { k1: v1, k2: v2, k3: v3, k4, v4, ... }
 // Gives priority to the objects which appear later in the list
-export const objMerge = <T>(objs: Record<string, T>[]) =>
-  objs.reduce((acc, obj) => ({ ...acc, ...obj }), {});
+export const objMerge = <T>(objs: T[]) => objs.reduce((acc, obj) => ({ ...acc, ...obj }), {} as T);
```

Comment:
This won't give you the type you want. tbh, I think this should just be removed and the usages replaced with `Object.assign({}, ...objs)` but if you want to keep it:
```suggestion
type UnionToIntersection<T> = (T extends any ? (x: T) => any : never) extends (
  x: infer R
) => any
  ? R
  : never;

export const objMerge = <T extends any[]>(
  objs: [...T]
): UnionToIntersection<T[number]> => Object.assign({}, ...objs);
```